napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -66,12 +73,10 @@ def setup_ui(self, error_message, readers):
         self.btn_box.rejected.connect(self.reject)
 
         # checkbox to remember the choice (doesn't pop up for folders with no extension)
-        if self._extension:
-            self.persist_checkbox = QCheckBox(
-                f'Remember this choice for files with a {self._extension} extension'
-            )
-            self.persist_checkbox.toggle()
-            layout.addWidget(self.persist_checkbox)
+        self.persist_checkbox = QCheckBox(self._persist_text)
+
+        self.persist_checkbox.toggle()
```

Comment:
```suggestion
        self.persist_checkbox.setChecked(True)
```

We should not depend on the state after call the constructor.